### ISSUE TITLE
Upload generated terrain-rgb to mapbox

### ISSUE THREAD

**Issue by gotestrand** [mapbox/rio-rgbify#19]
Hi,

I'm trying to achieve the same results as in the blogpost https://blog.mapbox.com/realistic-terrain-with-custom-styling-ce1fe98518ab

I've managed to create both geotif and .mbtiles from my terrain geotif, using rio-rgbify. After uploading the files to mapbox as a raster-layer the rgb-values get messed up by the compression of the tiles, causing the elevation-styling to be corrupt. 

I don't know if this is the right forum, but any ideas would be helpful!

**Comment by vincentsarago** [comment]
Hi @gotestrand thank, for opening the issue. 

> After uploading the files to mapbox as a raster-layer the rgb-values get messed up by the compression of the tiles, causing the elevation-styling to be corrupt.

I'm may be wrong but last time I checked we cannot use geotiff/mbtiles terrain rgb in studio because  when you upload it, it is recognized as `raster` and not as `raster-dem`. ref: https://github.com/mapbox/rio-rgbify/issues/17#issue-309403586

so it might result in something like 👇 
![capture d ecran 2018-06-29 a 10 00 44](https://user-images.githubusercontent.com/10407788/42096497-7cb64fe4-7b83-11e8-9f92-8292b573131b.jpg)

I'm not aware this will change soon, you may contact mapbox support and ask about it.

**Comment by andrewharvey** [comment]
I still have my :crossed_fingers: that eventually you'll be able to upload your single band GeoTIFF DEM to Mapbox and it will automatically handle converting it to terrain-rgb as a `raster-dem`.

**Comment by gotestrand** [comment]
@dnomadb Thank you! After a bit of struggle I've succesfully managed to get my raster-dem working in mapbox-gl 😄 

The final piece that was missing was to translate my geotiff to `EPSG:3857` projection, before generating the final `mbtiles` file. The projection translation would be a nice feature to add to this repo! 

My final steps in summary:

- Translate geotiff to web mercator projection `EPSG:3857`. Using gdal with `gdalwarp -t_srs EPSG:3857 -r lanczos -wo SOURCE_EXTRA=1000 -co COMPRESS=LZW [input-file] [output-file]`
- Create mbtiles using `rio rgbify  -b -10000 -i 0.1 --max-z 12 --min-z 0 --format png [input-file] [output-mbtiles]`
- Create a `raster-dem` source in mapbox, using `pngraw` tiles. Add layer to map, tada!

![image](https://user-images.githubusercontent.com/6369674/42697525-a1d3b3d2-86bb-11e8-8387-0475c45cc41e.png)

**Comment by predictwise** [comment]
Hi @dnomadb ,

Thanks for your summary! The first two steps were going smoothly, but I was stuck at the third step. I didn't know how to generate `pngraw` tiles. I just uploaded `.mbtiles` file to Mapbox Studio and got a `tileset ID`. Then, I added layer to map with `tileset ID` and got the following result:

![Screenshot from 2020-01-09 16-55-10](https://user-images.githubusercontent.com/12490765/72082725-e4364580-3300-11ea-8f94-1a8adb61ff0b.png)

It seems that terrain is composed of many squares. It is very strange!
